src/main/java/dev/waterdog/player/HandshakeUtils.java

```diff
@@ -72,7 +72,9 @@ public static boolean validateChain(JsonArray chainArray) throws Exception {
             }
 
             if (lastKey != null) {
-                EncryptionUtils.verifyJwt(jwt, lastKey);
+                if (!EncryptionUtils.verifyJwt(jwt, lastKey)) {
+                    return false;
```

Comment:
If the encryption verification fails this is supposed to throw an exception which we can use to detect whether it was successful or not ( see [here](https://github.com/CloudburstMC/Protocol/blob/8ec21ae916ed6b1338246920aa1657dcb4099339/bedrock/bedrock-common/src/main/java/com/nukkitx/protocol/bedrock/util/EncryptionUtils.java#L105) )

Reply:
But false already says that the decryption did failed, but I can make it throw an exception

Reply:
Hmm I'm wondering why it has both a return value and can throw an exception. Probably returns false if the validation fails and throws an exception If the key is unusable or smth, I'll look into that later

Reply:
So should I remove this part or can you accept it like this?

Reply:
I guess I can approve here, returning false and an exception should both work.

Reply:
Returning false is correct.